server/src/com/cloud/api/ApiServer.java

```diff
@@ -430,8 +433,27 @@ public void handle(final HttpRequest request, final HttpResponse response, final
             if (!(responseType.equals(HttpUtils.RESPONSE_TYPE_JSON) || responseType.equals(HttpUtils.RESPONSE_TYPE_XML))) {
                 responseType = HttpUtils.RESPONSE_TYPE_XML;
             }
-
             try {
+                //verify that parameter is legit for passing via admin port
```

Comment:
Check if it makes sense to move this as a separate helper method. There are also other places in code that reads the annotation on the API commands and parameters. Check if some of them can be reused.